android/src/main/java/com/mattermost/networkclient/Extensions.kt

```diff
@@ -46,3 +48,20 @@ fun Request.Builder.applyHeaders(headers: ReadableMap?): Request.Builder {
 
     return this;
 }
+
+/**
+ * Trims trailing slashes in the string
+ */
+fun String.trimTrailingSlashes(): String {
+    return trimEnd { c -> c == '/' }
+}
+
+/**
+ * Computes the MD5 of a string
+ */
+fun String.md5(): String {
+    val md = MessageDigest.getInstance("MD5")
```

Comment:
@jupenur Would you be able to provide a reason why? I'm very much interested in the answer for this.

Reply:
In this case, primarily because every time either we or customers of ours throw the code at some security automation, the automation will complain. There's no good reason to use MD5 other than it being already there in legacy code, so simply not using it makes everyone's lives a little bit easier.

There is a potential concern around hash collisions, too, but how realistic it is depends on how the library is used in the end. Basically, if a new NetworkClient can be created from attacker-controlled input, it's possible an attacker could craft a hash collision which would allow them to mess with the data of other NetworkClients.